task.go

```diff
@@ -325,7 +325,16 @@ func (t *task) Delete(ctx context.Context, opts ...ProcessDeleteOpts) (*ExitStat
 		return nil, fmt.Errorf("task must be stopped before deletion: %s: %w", status.Status, errdefs.ErrFailedPrecondition)
 	}
 	if t.io != nil {
-		t.io.Close()
+		// io.Wait locks for restored tasks on Windows unless we call
+		// io.Close first (https://github.com/containerd/containerd/issues/5621)
+		// in other cases, preserve the contract and let IO finish before closing
+		if t.client.runtime == fmt.Sprintf("%s.%s", plugin.RuntimePlugin, "windows") {
```

Comment:
It just occurred to me that this checks for `plugin.RuntimePlugin` (so does code further up); should these also check for  `plugin.RuntimeV2` ? And related to that; `DefaultRuntime`? Which is different on Windows;
https://github.com/containerd/containerd/blob/ee2c8b79bfde9039ba91aba7affa1abb3a3eea35/defaults/defaults_windows.go#L44C1-L45